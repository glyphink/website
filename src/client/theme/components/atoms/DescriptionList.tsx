/**
 * @file Components - DescriptionList
 * @module website/theme/components/atoms/DescriptionList
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type FRC,
  type JSX
} from 'react'
import type DescriptionListProps from './DescriptionList.props.mts'

/**
 * A description list.
 *
 * The component encloses a list of groups of terms (`DescriptionTerm`) and
 * details (`DescriptionDetails`).
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/dl
 * @see {@linkcode DescriptionListProps}
 * @see {@linkcode FRC}
 *
 * @category
 *  atoms
 *
 * @const {FRC<DescriptionListProps>} DescriptionList
 */
const DescriptionList: FRC<DescriptionListProps> = forwardRef<
  HTMLDListElement,
  DescriptionListProps
>(
  /**
   * @param {Props<DescriptionListProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLDListElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<dl>` element
   */
  (
    props: Props<DescriptionListProps>,
    ref: ForwardedRef<HTMLDListElement>
  ): JSX.Element => {
    return <dl {...props} ref={ref} />
  }
)

DescriptionList.displayName = 'DescriptionList'

export default DescriptionList
