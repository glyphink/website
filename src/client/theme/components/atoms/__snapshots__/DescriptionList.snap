// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/DescriptionList > should render <dl> 1`] = `
<dl>
  <dt>
    Beast of Bodmin
  </dt>
  <dd>
    A large feline inhabiting Bodmin Moor.
  </dd>
  <dt>
    Morgawr
  </dt>
  <dd>
    A sea serpent.
  </dd>
  <dt>
    Owlman
  </dt>
  <dd>
    A giant owl-like creature.
  </dd>
</dl>
`;
